src/main/java/com/project/cinema/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -0,0 +1,63 @@
+package com.project.cinema.dao.impl;
+
+import com.project.cinema.dao.ShoppingCartDao;
+import com.project.cinema.exceptions.DataProcessingException;
+import com.project.cinema.lib.Dao;
+import com.project.cinema.model.ShoppingCart;
+import com.project.cinema.model.User;
+import com.project.cinema.util.HibernateUtil;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class ShoppingCartDaoImpl implements ShoppingCartDao {
+
+    @Override
+    public ShoppingCart add(ShoppingCart shoppingCart) {
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            transaction = session.beginTransaction();
+            session.save(shoppingCart);
+            transaction.commit();
+            return shoppingCart;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert Shopping Cart entity", e);
+        }
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<ShoppingCart> query = criteriaBuilder.createQuery(ShoppingCart.class);
+            Root<ShoppingCart> root = query.from(ShoppingCart.class);
+            Predicate predicate = criteriaBuilder.equal(root.get("user"), user.getId());
+            CriteriaQuery<ShoppingCart> criteriaQuery = query.where(predicate);
+            return session.createQuery(criteriaQuery).uniqueResult();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't find any user", e);
+        }
+    }
+
+    @Override
+    public void update(ShoppingCart shoppingCart) {
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            transaction = session.beginTransaction();
+            session.update(shoppingCart);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't update Shopping Cart entity", e);
+        }
```

Comment:
agreed in operation with transaction id not to use try-with resources

Reply:
done